examples/di/InfiAgent-DABench/DABench.py

```diff
@@ -0,0 +1,174 @@
+
+
+def evaluate_accuracy_proportional_by_sub_question_adjusted(results):
+    total_score = 0
+    for result in results:
+        if "correctness" in result:
+            sub_question_count = len(result["correctness"])
+            score_per_sub_question = 1 / sub_question_count if sub_question_count > 0 else 0
+            question_score = sum(result["correctness"].values()) * score_per_sub_question
+            total_score += question_score
+    return round(total_score / len(results), 4) if results else 0
+
+
+class DABench:
+    def __init__(
+        self,
+        questions_file=Path(DABENCH_PATH) / "da-dev-questions.jsonl",
+        answers_file=Path(DABENCH_PATH) / "da-dev-labels.jsonl",
+        template="",
+    ):
+        # Read questions from a JSONL file
+        with open(questions_file, "r") as file:
+            self.questions = {int(json.loads(line)["id"]): json.loads(line) for line in file}
+
+        # Read answers from a JSONL file
+        with open(answers_file, "r") as file:
+            self.answers = {int(json.loads(line)["id"]): json.loads(line) for line in file}
+
+        self.template = template if template else DABENCH
+
+    def get_question(self, question_id):
+        """Retrieve the question by its id."""
+        return self.questions.get(question_id, "Question not found.")
+
+    def get_prompt(self, question_id):
+        """Retrieve the question by its id."""
+        temp = self.get_question(question_id)
+        return self.template.format(
+            question=temp["question"],
+            constraints=temp["constraints"],
+            format=temp["format"],
+            file_name=str(DABENCH_PATH) + "/da-dev-tables/" + temp["file_name"],
+            level=temp["level"],
+        )
+
+    def get_answer(self, answer_id):
+        """Retrieve the answer list by its id."""
+        return self.answers.get(answer_id, "Answer not found.")
+
+    def eval(self, id, prediction):
+        """Evaluate the prediction against the true label."""
+        true_label = self.get_answer(id)["common_answers"]
+        # Parse the prediction string into a dictionary of metric-value pairs
+        pred_dict = {}
+        for pred in prediction.split("@"):
+            if pred == "":
+                continue
+            parts = pred.strip().split("[")
+            metric = parts[0].strip().replace(",", "")
+            value = parts[1].replace(",", "").replace("]", "")
+            try:
+                value = float(value)
+            except:
+                value = value
+            pred_dict[metric] = value
+
+        # Sort the true labels to match the order of predictions
+        sorted_true_label = sorted(true_label, key=lambda x: x[0])
+
+        # Compare each prediction with the corresponding true label
+        correct = True
+        for metric, true_value in sorted_true_label:
+            try:
+                true_value = float(true_value)
+            except:
```

Comment:
Used too many try / except. But this is not our style.

Reply:
done